src/libraries/System.Private.CoreLib/src/System/Collections/Generic/HashSet.cs

```diff
@@ -811,8 +811,8 @@ public bool SetEquals(IEnumerable<T> other)
                 }
 
                 // Already confirmed that the sets have the same number of distinct elements, so if
-                // one is a superset of the other then they must be equal.
-                return ContainsAllElements(otherAsSet);
+                // one is a subset of the other then they must be equal.
+                return IsSubsetOfHashSetWithSameComparer(otherAsSet);
```

Comment:
This could end up having very different perf characteristics. ContainsAllElements is O(OtherCount) whereas IsSubsetOfHashSetWithSameComparer is O(ThisCount).  This change is changing which set is enumerated.  Unless we want to do a full analysis of common usage, this should be `otherAsSet.IsSubsetOfHashSetWithSameComparer(this)` in order to maintain the same behavior.

Reply:
@stephentoub in this case we know that ThisCount == OtherCount though, right (see check on 808)?

So I think it doesn't matter, except that it will impact which set instance is enumerated and which has Contains called on it.

Reply:
Ah, yes, that was hidden in the diff, and I was thinking this was the superset case.